### PR TITLE
Battery indicator support

### PR DESCRIPTION
**Pull request by shermp**
This PR adds support for displaying the current battery level. This can either be a label with a user defined template, the battery icon, or both together.

Current limitations are that you can not have both the clock and battery level in same placement and position. I tried to allow that, but was causing issues.

The battery icon does not display properly in dark mode, there doesn't appear to be an inverted version of the icon.

When displaying both level and icon together, there is a fairly large gap between them. I haven't managed to reduce that gap.

The settings file format has changed a lot, if you just want to test, take a backup of the old settings, as they are migrated then deleted.

### CODE REVIEW

File: src/nickelclock.cc [hunk 1]
```diff
@@ -12,89 +12,16 @@
...
 
-void NCSettings::setMaxHMargin(QRect const& screenGeom)
-{
-    int w = screenGeom.width() < screenGeom.height() ? screenGeom.width()
-                                                     : screenGeom.height();
-    maxHMargin = w / 4;
-    nh_log("screen width: %d, setting margin: %d", w, maxHMargin);
-}
+const char nc_sysfs_gen_bat_cap[] = "/sys/class/power_supply/battery/capacity";
+const char nc_sysfs_mc13892_bat_cap[] = "/sys/class/power_supply/mc13892_bat/capacity";
```
Comment (shermp) [inline src/nickelclock.cc:24]: @NiLuJe are these the only two paths I have to worry about?

File: src/nickelclock.cc [hunk 2]
```diff
@@ -242,27 +214,96 @@ void NC::setFooterStylesheet(ReadingFooter *rf)
...
+    if (bcFile.open(QIODevice::ReadOnly | QIODevice::Text)) {
+        bool ok = false;
+        battery = bcFile.readAll().trimmed().toInt(&ok);
+        if (!ok) {
+            nh_log("failed to get battery level");
+            battery = 100;
+        }
+    }
+    return battery;
+}
```
Comment (shermp) [inline src/nickelclock.cc:282]: @NiLuJe do you see any issues here?

File: src/nickelclock.cc (hunk 1 above)
Comment (NiLuJe) [reply src/nickelclock.cc:24]: So far, yeah ;).

File: src/nickelclock.cc (hunk 2 above)
Comment (NiLuJe) [reply src/nickelclock.cc:282]: Not entirely familiar with `QFile`, so I'm not sure if that's necessary here, but I don't see it getting `close`d?

Comment (NiLuJe) [reply src/nickelclock.cc:282]: (Possibly cache the `exists` test result on startup, as it's never going to change at runtime ;)).

Comment (shermp) [reply src/nickelclock.cc:282]: I thought I read in the documentation that QFile closes on destruction, but I just reread the documentation, and it doesn't say anything about it so I probably do need to close it. Good catch thank you.

Comment (shermp) [reply src/nickelclock.cc:282]: Oh, hang on, yes it does:

> [virtual]QFile::~QFile()
> Destroys the file object, closing it if necessary.

**Comment by shermp** [comment]
Issues with earlier firmware versions (4.23 is now known to work) resolved.

**Pull request merged**